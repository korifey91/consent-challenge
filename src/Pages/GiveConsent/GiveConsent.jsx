import {
  Button,
  Checkbox,
  FormControlLabel,
  FormGroup,
  TextField,
} from '@material-ui/core';

import styles from './GiveConsent.module.scss';

const consentsList = [
  {
    title: 'Receive newsletter',
    id: 1,
  },
  {
    title: 'Be shown targeted ads',
    id: 2,
  },
  {
    title: 'Contribute to anonymous visit statistics',
    id: 3,
  },
];

function GiveConsent() {
  const handleChange = () => {};
  return (
    <form className={styles.consentForm} noValidate autoComplete="off">
      <div>
        <TextField className={styles.input} label="Name" variant="outlined" />
        <TextField
          className={styles.input}
          label="Email address"
          variant="outlined"
        />
      </div>
      <p>I agree to:</p>
      <FormGroup className={styles.consents}>
        {consentsList.map((consent) => (
          <FormControlLabel
            key={consent.id}
            control={
              <Checkbox checked onChange={handleChange} color="primary" />
            }
            label={consent.title}
          />
        ))}
      </FormGroup>
      <Button className={styles.submit} variant="contained" color="primary">
        Give consent
      </Button>
    </form>
  );
}

export default GiveConsent;
