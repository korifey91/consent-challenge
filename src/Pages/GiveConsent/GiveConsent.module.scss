.consentForm {
  display: flex;
  flex-direction: column;
  text-align: center;
}

.input {
  margin: 10px;
}

.consents {
  border: 1px solid #c4c4c4;
  padding: 10px;
}

.submit {
  align-self: center;
  margin-top: 10px;
}
