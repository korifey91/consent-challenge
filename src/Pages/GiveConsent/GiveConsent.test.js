import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { unmountComponentAtNode } from 'react-dom';

import configureStore from 'redux-mock-store';
import { fireEvent, render, screen } from '@testing-library/react';

import GiveConsent from './GiveConsent';

const mockStore = configureStore([]);

let container = null;
let store = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);

  store = mockStore({
    consents: {
      data: [],
      meta: {
        totalCount: null,
        totalPages: null,
        currentPage: null,
      },
      error: null,
    },
  });
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

test('renders GiveConsent correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <GiveConsent />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('submit button disabled if all inputs are empty', () => {
  render(
    <Provider store={store}>
      <GiveConsent />
    </Provider>
  );

  expect(screen.getByRole('button')).toBeDisabled();
});

test('submit button is not disabled if all inputs fullfield', () => {
  render(
    <Provider store={store}>
      <GiveConsent />
    </Provider>
  );

  const inputs = screen.getAllByRole('textbox');
  inputs.forEach((input) =>
    fireEvent.change(input, {
      target: { value: 'JavaScript' },
    })
  );

  const checbox = screen.getByRole('checkbox', { name: 'Receive newsletter' });
  fireEvent.click(checbox);

  expect(screen.getByRole('button')).toBeEnabled();
});
