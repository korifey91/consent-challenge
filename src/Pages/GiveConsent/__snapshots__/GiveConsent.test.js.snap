// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GiveConsent correctly 1`] = `
<form
  autoComplete="off"
  className="consentForm"
  noValidate={true}
>
  <div
    className="inputs"
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
      >
        Name
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          name="name"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
      >
        Email address
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          name="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Email address
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <p>
    I agree to:
  </p>
  <div
    className="MuiFormGroup-root consents"
  >
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input-8"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Receive newsletter
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input-8"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Be shown targeted ads
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input-8"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Contribute to anonymous visit statistics
      </span>
    </label>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained submit MuiButton-containedPrimary Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Give consent
    </span>
  </button>
</form>
`;
